sklearn/metrics/pairwise.py

```diff
@@ -811,6 +811,9 @@ def cosine_similarity(X, Y=None):
     Y : array_like, sparse matrix (optional)
         with shape (n_samples_Y, n_features).
 
+    dense_output : boolean (optional)
+        Whether to return sparse output.
```

Comment:
Actually this is backwards. It should say 'Whether to return dense output.' Or better 'Whether to return dense output even when the input is sparse.  If 'False', the output is sparse if both input arrays are sparse.`